Hillshaderenderer: Add 'multidirectional oblique-weighted' rendering option

AsgerPetersen opened the pull request:
Adds 
- a new rendering method to the hillshaderenderer
- a checkbox to the gui to enable this method instead of the `classic` method
- XML serialization and deserialization

Serialized XML is using an attribute named `multidirection`. This might not be optimal, if we expect more methods to come.

AsgerPetersen committed:
Hillshaderenderer: Add 'multidirectional oblique-weighted' rendering option

nyalldawson commented on src/core/raster/qgshillshaderenderer.cpp:
```diff
@@ -106,6 +111,12 @@ QgsRasterBlock *QgsHillshadeRenderer::block( int bandNo, const QgsRectangle &ext
   double sinZenithRad = sin( zenithRad );
 
+  // Multi direction hillshade: http://pubs.usgs.gov/of/1992/of92-422/of92-422.pdf
+  double angle0_rad = (-1 * mLightAzimuth - 45 - 45 * 0.5) * M_PI / 180.0;
```
Use camelCase for QGIS variable names

nyalldawson commented on src/core/raster/qgshillshaderenderer.h:
```diff
@@ -102,11 +108,18 @@ class CORE_EXPORT QgsHillshadeRenderer : public QgsRasterRenderer
+     * @brief Set IsMultiDirection
+     * @param isMultiDirection Is multi direction
+     */
+    void SetMultiDirection( bool isMultiDirection ){ mMultiDirection = isMultiDirection; }
```
Should be `setMultiDirection`. But perhaps `setMultiDirectional` is a better name?

AsgerPetersen replied:
Or we could use a `setMethod( methodEnum )` to prepare for future methods? This would of course require changes to more parts of the code.

NathanW2 replied:
Don't stress about that too much easy to fix later if needed

nyalldawson commented:
Looks good! A couple of things to change:

- any public members added to the headers need to have the changes also added to the corresponding sip python bindings file under the python folder. This allows PyQGIS scripts to call the methods.

- compile with the option "with_astyle" on, and then run scripts/prepare_commit.sh prior to committing changes. This script ensures consistent whitespace and formatting throughout the code base.

Otherwise, nice change!

AsgerPetersen commented:
@nyalldawson Thank you for the review.

I have
- changed variable names to camelCase
- Renamed multidirection to multidirectional
- added sip bindings
- run `prepare_commit.sh`. (It didn't seem to change anything)

NathanW2 commented:
Thanks for all your great work on this.

AsgerPetersen commented:
@NathanW2 And thank you for your support!

NathanW2 commented:
Always welcome.  And a thanks to @nyalldawson for doing a quick code review.
